sqlglot/dialects/dialect.py

```diff
@@ -351,6 +351,9 @@ class Dialect(metaclass=_Dialect):
         - Clickhouse, which will forward the alias across the query i.e it resolves to "WHERE id = 1 GROUP BY id HAVING id = 1"
     """
 
+    EXPAND_ALIAS_REFS_ONLY_IN_GROUP_BY = False
+    """Whether alias reference expansion should only happen for the GROUP BY clause."""
+
```

Comment:
I think we should add the `EARLY` prefix somewhere in this flag e.g. `EARLY_EXPAND_ALIAS_REFS_IN_GROUPBY` as this is the differentiator in the original PR.

If `schema` is non-empty we will revisit the 2nd `_expand_alias_refs()` for the rest of the expansion logic, so maybe `ONLY_IN_GROUP_BY` can be confusing (?)